### PR TITLE
Fix Gmax Pokemon learnsets in the teambuilder

### PR DESCRIPTION
**Pull request by KrisXV**

### CODE REVIEW

File: src/battle-dex-data.ts [hunk 1]
```diff
@@ -1370,7 +1370,7 @@ class Species implements Effect {
 		this.battleOnly = data.battleOnly || (this.isGigantamax ? this.baseSpecies : undefined);
 		this.isNonstandard = data.isNonstandard || null;
 		this.unreleasedHidden = data.unreleasedHidden || false;
-		this.changesFrom = data.changesFrom || undefined;
+		this.changesFrom = data.changesFrom || (this.isGigantamax ? this.baseSpecies : undefined);
```
Comment (Zarel) [inline src/battle-dex-data.ts:1373]: The `battleOnly` check should cover it; it 100% does not need to be repeated here.

Comment (KrisXV) [reply src/battle-dex-data.ts:1373]: It doesn't.

Comment (Zarel) [reply src/battle-dex-data.ts:1373]: Then it should be fixed in the implementation, not here.

Comment (Zarel) [reply src/battle-dex-data.ts:1373]: Like you didn't update the documentation in `FORMES.md`, nor the jsdoc in `dex-data.ts`. Those are things you _must_ do if you want to change what a property means.

But at some point you should probably just realize that you don't actually want to change what the property means, but just how it's used.

**Pull request closed without merging**